src/nupic/algorithms/KNNClassifier.py

```diff
@@ -361,7 +361,9 @@ def learn(self, inputPattern, inputCategory, partitionId=None, isSparse=0,
     @param inputCategory (int) The category to be associated to the training
         pattern
 
-    @param partitionId (int) UNKNOWN
+    @param partitionId (int) During inference all training vectors with
+        partitionId equal to that of the test vector are ignored.
+        See description of parameter at infer() for further details.
```

Comment:
We should have a helpful higher level description of what partition ID is because it actually has multiple uses. Something like "partitionID allows you to partition the data set and associate unique IDs with sets of vectors. One use case is to ignore a specific set of vectors during inference for k-fold cross validation (see description of infer() for further details). This is an optional parameter."